include/zmq.h

```diff
@@ -371,6 +374,18 @@ ZMQ_EXPORT const char *zmq_msg_gets (const zmq_msg_t *msg, const char *property)
 #define ZMQ_VMCI_BUFFER_MAX_SIZE 87
 #define ZMQ_VMCI_CONNECT_TIMEOUT 88
 #define ZMQ_USE_FD 89
+#define ZMQ_PEER_EVENTS 90
+
+/*  getsockopt additional struct to recover peer events                       */
+typedef struct zmq_gso_peer_events_t
+{
+    void *zid;
```

Comment:
What does "zid" mean? I think the ZMQ API uses the term "identity" as in ZMQ_IDENTITY. This might be prefixed here with "peer", i.e. peer_identity and peer_identity_len
This applies to all other uses of "zid" as well.

(I am not so sure with _len vs. _size; "len" might suggest the identity is a string, which it is not)

Reply:
zid = ZMQ Identity

id is to much overloaded, pid (peer id) was too confusing with posix PID. If you have a better name for it, I am open to change it.

_len vs _size; I can change to _size, if that's making it clear.

Reply:
Again, using an abbreviation here is not a good idea at all IMO.

zid/ZMQ identity is confusing, since this prefix is not used for any other field.

The use of the term "identity" may be confusing (I think this was discussed somewhere else), but it is the term the ZMQ API uses. Using another term at some places would be even more confusing.

Reply:
I think rounting_id might be a solution. It is something that started being used instead of identity few years ago around zmq projects. We can continue and use it (zproto, czmq, SERVER socket type)

Reply:
Ok... but then ZMQ_ROUTING_ID should be added as a synonym for ZMQ_IDENTITY, and ZMQ_IDENTITY should be declared deprecated.

Reply:
sounds good to me

Reply:
Hm, and what about ZMQ_MSG_PROPERTY_IDENTITY and its value, "Identity"?

Reply:
Eh... There already is a pipe_t::routing_id with another meaning. This would need to be renamed to avoid a conflict.

Reply:
Please clarify, whn you reach a consesus

Reply:
@somdoron Your comment was somehow placed outside this thread... You said it is the same meaning. That cannot be true. pipe_t::routing_id and router_t::next_rid are both uint32_t, rather than a blob. pipe_t has members routing_id and identity, so I cannot rename identity to routing_id, since this already names another member.

Reply:
@msune I think we agreed on "routing_id" :) Sorry for the confusion.